tests/functional/NotificationTargetKnowbaseItem.php

```diff
@@ -0,0 +1,147 @@
+        foreach ($knowbasenotifs as $kbnotif) {
+            $datawant = $kbnotif;
+            $datawant['is_active'] = 1;
+            $notif->update($datawant);
+            $kbnotif = $notif->getById($kbnotif['id'])->fields;
+            $kbnotif['date_mod'] = null;
+            $this->array($kbnotif)->isIdenticalTo($datawant);
+        }
+        //search glpi user
+        $email = new \UserEmail();
+        $this->integer($email->add(
+            [
+                'users_id' => getItemByTypeName('User', 'glpi', true),
+                'email' => 'test@test.com',
+                'is_default' => 1,
+            ]
+        ))->isGreaterThan(0);
+
+        // test create group
+        $groups = new Group();
+        $groupid = $groups->add(
+            [
+                'name' => 'testknowbasegroup',
+            ]
+        );
+        $this->integer($groupid)->isGreaterThan(0);
+        $group = $groups->getById($groupid)->fields;
+        $this->string($group['name'])->isEqualTo('testknowbasegroup');
+
+        //add user to group
+        $groupsusers = new \Group_User();
+        $groupsusers->add(
+            [
+                'groups_id' => $group['id'],
+                'users_id' => getItemByTypeName('User', 'glpi', true),
+            ]
+        );
+
+        //test add group for notification
+        $notiftarget = new NotificationTarget();
+        foreach ($knowbasenotifs as $kbnotif) {
+            $ntargetid = $notiftarget->add(
+                [
+                    'notifications_id' => $kbnotif['id'],
+                    'type' => Notification::GROUP_TYPE,
+                    'items_id' => $group['id'],
+                ]
+            );
+            $ntargetdata = $notiftarget->getById($ntargetid)->fields;
+            $this->array($ntargetdata)->isIdenticalTo(
+                [
+                    'id' => $ntargetid,
+                    'items_id' => $group['id'],
+                    'type' => Notification::GROUP_TYPE,
+                    'notifications_id' => $kbnotif['id']
+                ]
+            );
+        }
+
+        //create/update/delete knowbase item
+        $knowbaseitem = new \KnowbaseItem();
+        $id = $knowbaseitem->add(
+            [
+                'name' => 'testknowbaseitem',
+                'answer' => 'testknowbaseitem',
+                'users_id' => getItemByTypeName('User', 'glpi', true),
+            ]
+        );
+        $knowbaseitem->update(
+            [
+                'id' => $id,
+                'name' => 'testknowbaseitemupdate',
+                'answer' => 'testknowbaseitemupdate',
+                'users_id' => getItemByTypeName('User', 'glpi', true),
+            ]
+        );
+        $knowbaseitem->delete(['id' => $id]);
+
+        //test check if is in notification queue
+        $notifqueue = new QueuedNotification();
+        $count = count($notifqueue->find(['itemtype' => 'KnowbaseItem']));
+        $this->integer($count)->isEqualTo(3);
```

Comment:
You should check the notification queue after each action, and ensure that the sent notification, at least, corresponds to the expected event.